src/store/state.ts

```diff
@@ -49,6 +49,9 @@ const initialState: RootState = {
   experiments: {},
   watsNewModalVersion: '',
   enabledChains: {},
+  nftAssets: {},
+  nftAssetsNumber: 0,
+  starredNFTs: []
```

Comment:
There seems to be a lot of complexity introduced because of this array which is separately storing the starred NFTs. It would be better if `starred` was just a property of the NFT, to get a list of starred NFTs, you have to just filter that and that can be done on the UI